python_modules/libraries/dagster-sigma/dagster_sigma/resource.py

```diff
@@ -389,10 +399,30 @@ def build_defs(
         Returns:
             Definitions: The set of assets representing the Sigma content in the organization.
         """
-        with self.process_config_and_initialize_cm() as initialized_organization:
-            return SigmaOrganizationDefsLoader(
+        return Definitions(assets=load_sigma_asset_specs(self, dagster_sigma_translator))
+
+
+def load_sigma_asset_specs(
+    organization: SigmaOrganization,
+    dagster_sigma_translator: Type[DagsterSigmaTranslator] = DagsterSigmaTranslator,
```

Comment:
One last thing here. I think we should standarize on a translator _instance_ not a type because that will give users more flexibility.

Reply:
The benefit of using a type here is that we can instantiate a translator to hold the "global state", which is useful for example in implementing get_asset_spec() in the case that we need to e.g. build an asset key for an upstream asset: https://github.com/dagster-io/dagster/blob/master/python_modules/libraries/dagster-powerbi/dagster_powerbi/translator.py#L151

Can you think of an alternative way to do this?

Reply:
I'm not sure I follow fully - are you suggesting typing the translator as a callable and/or dynamically creating translator classes after binding the "load context"?

> I think we should standarize on a translator instance not a type because that will give users more flexibility.
What flexibility do you anticipate moving from translator type to instance enabling? Could users accomplish this with a class/type factory?

Reply:
Yeah they could it with a class factory, but that just is more "goofy" for lack of a better term.